src/libcollections/str.rs

```diff
@@ -877,19 +898,19 @@ impl str {
     /// # Examples
     ///
     /// ```
-    /// let four_lines = "foo\nbar\n\nbaz";
+    /// let four_lines = "foo\r\nbar\n\nbaz";
     /// let v: Vec<&str> = four_lines.lines().collect();
     ///
-    /// assert_eq!(v, ["foo", "bar", "", "baz"]);
+    /// assert_eq!(v, ["foo\r", "bar", "", "baz"]);
     /// ```
     ///
     /// Leaving off the trailing character:
     ///
     /// ```
-    /// let four_lines = "foo\nbar\n\nbaz\n";
+    /// let four_lines = "foo\r\nbar\n\nbaz\n";
     /// let v: Vec<&str> = four_lines.lines().collect();
     ///
-    /// assert_eq!(v, ["foo", "bar", "", "baz"]);
+    /// assert_eq!(v, ["foo\r", "bar", "", "baz"]);
```

Comment:
Would be best not to document this if it's going to change. Maybe it's unlikely, but there's a lines() rfc coming (mostly focused on BufRead::lines though).